docs/src/content/en/docs/agents/overview.mdx

```diff
@@ -54,264 +78,237 @@ export const mastra = new Mastra({
 
-For more real-time responses, you can stream the agent's response:
+### Streaming text
+
+Call `.stream()` with an array of message objects that include `role` and `content`:
 
 ```typescript showLineNumbers copy
 const stream = await testAgent.stream([
-  { role: "user", content: "Tell me a story." },
+  { role: "user", content: "Help me organize my day" },
+  { role: "user", content: "My day starts at 9am and finishes at 5.30pm" },
+  { role: "user", content: "I take lunch between 12:30 and 13:30" },
+  { role: "user", content: "I have meetings Monday to Friday between 10:30 and 11:30" }
 ]);
 
-console.log("Agent:");
-
 for await (const chunk of stream.textStream) {
   process.stdout.write(chunk);
 }
 ```
 
-For more details about streaming responses, see the [stream reference documentation](/reference/agents/stream).
+### Completion using `onFinish()`
 
-## 3. Describing images
-
-Agents can analyze and describe images by processing both the visual content and any text within them. To enable image analysis, pass an object with `type: 'image'` and the image URL in the content array. You can combine image content with text prompts to guide the agent's analysis.
+When streaming responses, the `onFinish()` callback runs after the LLM finishes generating its response and all tool executions are complete.
+It provides the final `text`, execution `steps`, `finishReason`, token `usage` statistics, and other metadata useful for monitoring or logging.
 
 ```typescript showLineNumbers copy
-await testAgent.generate([
-  {
-    role: 'user',
-    content: [
-      {
-        type: 'image',
-        image: "https://example.com/images/test-image.jpg",
-        mimeType: "image/jpeg"
-      },
-      {
-        type: 'text',
-        text: 'Describe the image in detail, and extract all the text in the image.',
-      },
-    ],
-  },
-]);
+const stream = await testAgent.stream("Help me organize my day", {
+  onFinish: ({ steps, text, finishReason, usage }) => {
+    console.log({ steps, text, finishReason, usage });
+  }
+});
+
+for await (const chunk of stream.textStream) {
+  process.stdout.write(chunk);
+}
 ```
 
-## 4. Structured Output
+## Structured output
 
-Agents can return structured data by providing a JSON Schema or using a Zod schema.
+Agents can return structured, type-safe data by defining the expected output with either a [JSON Schema](https://json-schema.org/) or a [Zod](https://zod.dev/) schema. In both cases, the parsed result is available on `response.object`, allowing you to work directly with validated and typed data.
 
 ### Using JSON Schema
 
-```typescript showLineNumbers copy
-const schema = {
-  type: "object",
-  properties: {
-    summary: { type: "string" },
-    keywords: { type: "array", items: { type: "string" } },
-  },
-  additionalProperties: false,
-  required: ["summary", "keywords"],
-};
+Define the `output` shape using [JSON Schema](https://json-schema.org/):
```

Comment:
I wonder if we should even have a json schema section here since it's harder and not as nice as writing a zod schema. Perhaps just mentioning it's possible (above) is enough

Reply:
Yeah, that's a fair point!